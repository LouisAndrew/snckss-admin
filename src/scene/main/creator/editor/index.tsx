import React, { useState, useEffect } from 'react'
import { useFirestore } from 'reactfire'

import { Creations } from '..'
import Categories from '../../../../components/creator-pages/categories'
import Brands from '../../../../components/creator-pages/brands'
import { Category } from 'ts/interfaces/category'
import { Brand } from 'ts/interfaces/brand'
import { Product } from 'ts/interfaces/product'

import './styles.scss'
import Products from 'components/creator-pages/products'

interface Props {
        nowCreating: Creations
}

// TODO: fetch new data everytime success is called
const MainEditor: React.FC<Props> = ({ nowCreating }) => {
        const [allCategories, setAllCategories] = useState<Category[]>([])
        const [allBrands, setAllBrands] = useState<Brand[]>([])
        const [allProducts, setAllProducts] = useState<Product[]>([])
        const [rerender, setRerender] = useState(false)

        const fs = useFirestore()
        const categoriesRef = fs.collection('categories')
        const brandsRef = fs.collection('brand')
        const productsRef = fs.collection('product')

        useEffect(() => {
                fetchAllCtgs()
                fetchAllBrds()
                fetchAllProds()
        }, [])

        useEffect(() => {
                if (rerender) {
                        setRerender(false)
                }
        }, [rerender])

        const fetchAllCtgs = () => {
                categoriesRef.onSnapshot((docs) => {
                        let total: Category[] = []
                        docs.forEach((doc) => {
                                const data: Category = doc.data() as Category
                                total = [...total, data]
                        })
                        setAllCategories(total)
                })
        }

        const fetchAllBrds = () => {
                brandsRef.onSnapshot((docs) => {
                        let total: Brand[] = []
                        docs.forEach((doc) => {
                                const data: Brand = doc.data() as Brand
                                total = [...total, data]
                        })
                        setAllBrands(total)
                })
        }

        const fetchAllProds = () => {
                productsRef.onSnapshot((docs) => {
                        let total: Product[] = []
                        docs.forEach((doc) => {
                                const data: Product = doc.data() as Product
                                total = [...total, data]
                        })
                        setAllProducts(total)
                })
        }

        const doRerender = () => {
                setRerender(true)
        }

        return (
                <>
                        {!rerender && nowCreating === Creations.CATEGORIES && (
                                <Categories
                                        allCategories={allCategories}
                                        allBrands={allBrands}
                                        doRerender={doRerender}
                                        allProducts={allProducts}
                                />
                        )}
                        {!rerender && nowCreating === Creations.BRAND && (
                                <Brands
                                        allBrands={allBrands}
                                        allProducts={allProducts}
                                        doRerender={doRerender}
                                />
                        )}
                        {!rerender && nowCreating === Creations.PRODUCT && (
                                <Products
                                        allCategories={allCategories}
                                        allBrands={allBrands}
                                        allProducts={allProducts}
                                        doRerender={doRerender}
                                />
                        )}
                </>
        )
}

export default MainEditor
