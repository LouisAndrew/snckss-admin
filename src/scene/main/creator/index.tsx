import React, { useState } from 'react'
import { Container, Col, Row, Nav } from 'reactstrap'

import Tabs from '../../../components/tabs'
import MainEditor from './editor'

export enum Creations {
        NONE,
        CATEGORIES,
        BRAND,
        PRODUCT,
}

interface Creation {
        text: string
        creationTag: Creations
        className: string
}

interface Props {}

const Creator: React.FC<Props> = ({}) => {
        // Creations.NONE
        const [nowCreating, setNowCreating] = useState(Creations.BRAND)

        // data of things going to be created.
        const creations: Creation[] = [
                {
                        text: 'Categories',
                        creationTag: Creations.CATEGORIES,
                        className: 'categories',
                },
                {
                        text: 'Brands',
                        creationTag: Creations.BRAND,
                        className: 'brands',
                },
                {
                        text: 'Products',
                        creationTag: Creations.PRODUCT,
                        className: 'products',
                },
        ]

        // handling click by setting state based on which link is being clicked
        const handleClick = (creation: Creations) => {
                setNowCreating(creation)
        }

        return (
                <Container>
                        <Row>
                                <Col m="4" lg="4" xl="4">
                                        <h3>Select what you want to create</h3>
                                        <Nav tabs vertical>
                                                {creations.map((cr) => {
                                                        const {
                                                                text,
                                                                creationTag,
                                                                className,
                                                        } = cr
                                                        return (
                                                                <Tabs
                                                                        key={
                                                                                cr.className
                                                                        }
                                                                        handleClick={() => {
                                                                                handleClick(
                                                                                        creationTag
                                                                                )
                                                                        }}
                                                                        className={
                                                                                className
                                                                        }
                                                                >
                                                                        {text}
                                                                </Tabs>
                                                        )
                                                })}
                                        </Nav>
                                </Col>
                                <Col m="8" lg="8" xl="8">
                                        {nowCreating !== Creations.NONE && (
                                                <MainEditor
                                                        data-testid="editor"
                                                        nowCreating={
                                                                nowCreating
                                                        }
                                                />
                                        )}
                                </Col>
                        </Row>
                </Container>
        )
}

export default Creator
