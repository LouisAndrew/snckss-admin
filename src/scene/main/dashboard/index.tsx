import React from 'react'
import { NavItem, NavLink, Nav } from 'reactstrap'

import { Views } from '..'
import './styles.scss'
import Tabs from '../../../components/tabs'

interface Props {
        changeView: (view: Views) => void
}

const Dashboard: React.FC<Props> = ({ changeView }) => {
        const tabs = [
                {
                        text: 'Create Products',
                        className: 'product',
                        clickLink: Views.CREATE,
                },
                {
                        text: 'Show Orders',
                        className: 'orders',
                        clickLink: Views.ORDERS,
                },
        ]

        return (
                <Nav tabs vertical className="tabs">
                        {tabs.map((tab) => (
                                <Tabs
                                        handleClick={() => {
                                                changeView(tab.clickLink)
                                        }}
                                        className={tab.className}
                                        key={tab.className}
                                >
                                        {tab.text}{' '}
                                </Tabs>
                        ))}
                </Nav>
        )
}

export default Dashboard
