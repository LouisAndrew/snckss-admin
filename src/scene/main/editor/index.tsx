import React from 'react'

interface Props {}

const Editor: React.FC<Props> = ({}) => {
        return (
                <>
                        <h1>Editor</h1>
                </>
        )
}

export default Editor
