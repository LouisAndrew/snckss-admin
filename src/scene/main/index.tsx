import React, { useState } from 'react'
import { Container, Row, Col } from 'reactstrap'

import Dashboard from './dashboard'
import Editor from './editor'
import NonePage from './none'
import Orders from './orders'

export enum Views {
        NONE,
        CREATE,
        ORDERS,
}

interface Props {}

const Main: React.FC<Props> = ({}) => {
        const [view, setView] = useState(Views.NONE)

        const changeView = (view: Views): void => {
                setView(view)
        }

        return (
                <Container fluid>
                        <Row>
                                <Col md="3" lg="3" xl="3">
                                        <Dashboard changeView={changeView} />
                                </Col>
                                <Col md="9" lg="9" xl="9">
                                        {view === Views.NONE ? (
                                                <NonePage />
                                        ) : view === Views.CREATE ? (
                                                <Editor />
                                        ) : (
                                                <Orders />
                                        )}
                                </Col>
                        </Row>
                </Container>
        )
}

export default Main
