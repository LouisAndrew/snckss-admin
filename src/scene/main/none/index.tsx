import React from 'react'

interface Props {}

const NonePage: React.FC<Props> = ({}) => {
        return (
                <>
                        <h1>Nope</h1>
                </>
        )
}

export default NonePage
