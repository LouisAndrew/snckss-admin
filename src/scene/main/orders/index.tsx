import React from 'react'

interface Props {}

const Orders: React.FC<Props> = ({}) => {
        return (
                <>
                        <h1>Orders!</h1>
                </>
        )
}

export default Orders
