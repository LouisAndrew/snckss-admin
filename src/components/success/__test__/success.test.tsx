import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import SuccessPage from '..'
import { Creations } from '../../../scene/main/creator'

describe('Succes reusable page.', () => {
        afterEach(cleanup)
        const elProducts: React.ReactElement = (
                <SuccessPage type={Creations.PRODUCT} create={true} />
        )
        const elProductsUpdate: React.ReactElement = (
                <SuccessPage type={Creations.PRODUCT} create={false} />
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(elProducts, div)
        })

        it('renders when given a type correctly', () => {
                const { getByTestId } = render(elProducts)
                const h2: HTMLHeadingElement = getByTestId(
                        'success'
                ) as HTMLHeadingElement

                expect(h2).toHaveTextContent('product')
        })

        it('renders when given a type and shows if user created a product correctly', () => {
                const { getByTestId } = render(elProducts)
                const h2: HTMLHeadingElement = getByTestId(
                        'success'
                ) as HTMLHeadingElement

                expect(h2).toHaveTextContent('adding new')
        })

        it('renders when given a type and shows if user updated a product correctly', () => {
                const { getByTestId } = render(elProductsUpdate)
                const h2: HTMLHeadingElement = getByTestId(
                        'success'
                ) as HTMLHeadingElement

                expect(h2).toHaveTextContent('updating existing')
        })

        it('matches snapshot', () => {
                const tree = renderer.create(elProducts).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
