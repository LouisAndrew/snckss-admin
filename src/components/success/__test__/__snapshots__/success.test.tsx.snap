// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Succes reusable page. matches snapshot 1`] = `
<div
  className="success row container-fluid"
>
  <div
    className="inner"
  >
    <h2
      data-testid="success"
    >
      Success
       adding new 
      product
    </h2>
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M9 20.42l-6.21-6.21l2.83-2.83L9 14.77l9.88-9.89l2.83 2.83L9 20.42z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
</div>
`;
